exchanges/coinbene/coinbene.go

```diff
@@ -0,0 +1,344 @@
+package coinbene
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/config"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/websocket/wshandler"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+// Coinbene is the overarching type across this package
+type Coinbene struct {
+	exchange.Base
+	WebsocketConn *wshandler.WebsocketConnection
+}
+
+const (
+	coinbeneAPIURL     = "http://openapi-exchange.coinbene.com/api/exchange/"
+	coinbeneAuthPath   = "/api/exchange/v2"
+	coinbeneAPIVersion = "v2"
+	buy                = "buy"
+	sell               = "sell"
+
+	// Public endpoints
+	coinbeneFetchTicker    = "/market/ticker/one"
+	coinbeneFetchOrderBook = "/market/orderBook"
+	coinbeneGetTrades      = "/market/trades"
+	coinbeneGetAllPairs    = "/market/tradePair/list"
+	coinbenePairInfo       = "/market/tradePair/one"
+
+	// Authenticated endpoints
+	coinbeneGetUserBalance = "/account/list"
+	coinbenePlaceOrder     = "/order/place"
+	coinbeneOrderInfo      = "/order/info"
+	coinbeneRemoveOrder    = "/order/cancel"
+	coinbeneOpenOrders     = "/order/openOrders"
+	coinbeneClosedOrders   = "/order/closedOrders"
+
+	authRateLimit   = 150
+	unauthRateLimit = 10
```

Comment:
I gave some rate limits earlier https://github.com/thrasher-corp/gocryptotrader/pull/366#discussion_r334303863 based on lowest amounts since they vary by endpoint and our rate limiter doesn't support that. However my QUICKMATH numbers were way off and these don't even match my provided numbers :D
lowest unauthenticated rate limit is 2 requests per second and lowest authenticated limit is 3 times per second

Reply:
in scotts comment: 20 per 2s -> 10 per s
and 5 per 2s -> 150 per 60s (1minute)